File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:   
        num = len(nums)
        for i in range(num):
            for j in range(i + 1, num):
                if (nums[i] == nums[j]):
                    return True
        
        return False
        

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java
class Solution {
    public int findDuplicate(int[] nums) {
        Set<Integer> values = new HashSet<>();
        int duplicate = 0;
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            if (values.contains(num)) {
                // return the value
                duplicate = num;
            } else {
                values.add(num);
            }
        }
        return duplicate;
    }
}
